### ISSUE TITLE
Can provider support multi serialization protocols?

### ISSUE THREAD

**Issue by RojerAlone** [apache/dubbo#5180]
I'm trying to change my service's serialization protocol gracefully, so I want to implement a feature like this:

provider can support multi serialization protocols, and consumer assigns its serialization protocol, provider can receive consumer's request correctlly if consumer's serialization protocol is supported by provider.

I found there's a issue #4640 related my problem, and known that the consumer can't assign self serizalization protocol.

I try to set the `dubbo.protocol.serialization = protostuff,fastjson` in the `application.yml`, the provider service can start up successfully and the registry url contains `serialization` field, but when I send request to the provider, I got an exception `No such extension org.apache.dubbo.common.serialize.Serialization by name protostuff,fastjson`, it seems provider can't support multi serialization protocols, but why the [code](https://github.com/apache/dubbo/blob/master/dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ProtocolConfig.java#L339) shows it seems it supports?

Am I using it wrong?

And more, I think it's not graceful enough that consumer can't assign self serialization protocol, can you introduce to me why design it like this?

**Comment by RojerAlone** [comment]
@zyna If the serialization protocol is only supported one, then the `setSerialization` method does't need to check if serialization is multi extension:

```java
    public void setSerialization(String serialization) {
        if (DUBBO_PROTOCOL.equals(name)) {
            checkMultiExtension(Serialization.class, Constants.SERIALIZATION_KEY, serialization);
        }
        this.serialization = serialization;
    }
```

**Comment by zyna** [comment]
@RojerAlone Yes, I think _checkMultiExtension_ method  is  not necessary,can use  _checkExtension_ method to replace it.

**Comment by zyna** [comment]
you can try like this.
<dubbo:protocol id="p1" name="dubbo" serialization="hessian2" />
<dubbo:protocol id="p2" name="dubbo" serialization="java"/>
<dubbo:service interface="XX" ref="XX" version="XX" protocol="p1,p2" />

**Comment by RojerAlone** [comment]
> you can try like this.
> <dubbo:protocol id="p1" name="dubbo" serialization="hessian2" />
> <dubbo:protocol id="p2" name="dubbo" serialization="java"/>
> <dubbo:service interface="XX" ref="XX" version="XX" protocol="p1,p2" />

Is this configuratiton indicates the provider supports multi protocol? so how consumers choose which to use. I use spring boot, it seems to unsupport multi protocol field in the yml file.
